src/rules/parameter-properties.ts

```diff
@@ -1,12 +1,110 @@
-import { createRule } from "../utils.js";
+import { AST_NODE_TYPES } from "@typescript-eslint/utils";
+import { isIdentifier } from "@typescript-eslint/utils/ast-utils";
+import { RuleFix } from "@typescript-eslint/utils/ts-eslint";
 
+import { createRule } from "../utils.js";
 export const rule = createRule({
 	create(context) {
 		return {
 			TSParameterProperty(node) {
+				const paramNode = node.parameter;
+				if (
+					paramNode.type !== AST_NODE_TYPES.Identifier &&
+					paramNode.type !== AST_NODE_TYPES.AssignmentPattern
+				) {
+					return;
+				}
+
+				const classBody = node.parent.parent.parent;
+				if (classBody?.type !== AST_NODE_TYPES.ClassBody) {
+					return;
+				}
+
+				const constructor = node.parent;
+				if (constructor.type !== AST_NODE_TYPES.FunctionExpression) {
+					return;
+				}
```

Comment:
These three `if` statements are still missing coverage. You can check this locally with `pnpm run test run --coverage` -> opening `coverage/index.html` in a browser. For each case, >=1 of the following 3 is happening:

* It's a good check for a real case that should be unit tested
* The case it checks for can't be hit, and:
  * Removing it can be done without issue
  * It's helpful for type narrowing - which would mean a type assertion should be used here, with a comment to a new or existing issue on typescript-eslint to improve the AST